bool flag
float32 obstacle_size
float64 distance_to_obstacle
float64[3] initial_position
